app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -104,7 +131,19 @@ def with_provider_connection(options = {})
       connection = connect(options)
       yield connection
     ensure
-      connection.try(:disconnect) rescue nil
+      # The `connect` method will return different types of objects depending on the value of the `version`
+      # parameter. If `version` is 3 the connection object is created by the `ovirt` gem and it is closed
+      # using the `disconnect` method. If `version` is 4 the object is created by the oVirt Ruby SDK, and
+      # it is clused using the `close` method.
+      begin
+        if connection.respond_to?(:disconnect)
+          connection.disconnect
+        elsif connection.respond_to?(:close)
+          connection.close
+        end
+      rescue
```

Comment:
this rescues every StandardError and discards it. 
can you be more specific on what to rescue? 
or at least log the error?

```ruby
rescue => error
  $rhevm_log.error("error while disconnecting #{error}")
end
```